docs/src/test-fixtures-js.md

```diff
@@ -722,3 +722,59 @@ export const test = base.extend({
   }, { title: 'my fixture' }],
 });
 ```
+
+## Adding global beforeEach/afterEach hooks
+
+[`method: Test.beforeEach`] and [`method: Test.afterEach`] hooks run before/after each test declared in the same file and same [`method: Test.describe`] block (if any). If you want to declare hooks that run before/after each test globally, you can declare them as auto fixtures with `scope: 'test'` like this:
+
+```ts title="fixtures.ts"
+import { test as base } from '@playwright/test';
+
+export const test = base.extend({
+  forEachTest: [async ({ page, baseURL }, use) => {
+    await page.goto('http://localhost:8000');
+    await use();
+    console.log('Last URL:', page.url());
+  }, { scope: 'test', auto: true } ],  // starts automatically for every test, we pass "auto" for that.
+});
+```
+
+And then import the fixtures in all your tests:
+
+```ts title="mytest.spec.ts"
+import { test } from './fixtures';
+import { expect } from '@playwright/test';
+
+test('basic', async ({ page, baseURL }) => {
+  expect(page).toHaveURL(baseURL!);
+});
+```
+
+## Adding global beforeAll/afterAll hooks
+
+[`method: Test.beforeAll`] and [`method: Test.afterAll`] hooks run before/after all tests declared in the same file and same [`method: Test.describe`] block (if any), once per worker process. If you want to declare hooks
+that run before/after all tests in every file, you can declare them as auto fixtures with `scope: 'worker'` as follows:
+
+```ts title="fixtures.ts"
+import { test as base } from '@playwright/test';
+
+export const test = base.extend({
+  forEachWorker: [async ({ browser }, use) => {
+    console.log('Before All', browser.version());
```

Comment:
Users are often confused why Before All runs more than once, this only adds to the confusion. At the very least, say "`Before worker #${workerId}`"